mysys/crc32/crc32c_amd64.cc

```diff
@@ -184,9 +184,9 @@ static inline uint64_t CombineCRC(
 // Compute CRC-32C using the Intel hardware instruction.
 extern "C"
 USE_PCLMUL
-uint32_t crc32c_3way(uint32_t crc, const char *buf, size_t len)
+uint32_t crc32c_3way(uint32_t crc, const void *buf, size_t len)
 {
-  const unsigned char* next = (const unsigned char*)buf;
+  const unsigned char* next = static_cast<const unsigned char*>(buf);
```

Comment:
I do  not think changing this line is necessary. It is , for the most part, not our code anyway, so l'd leave it as-is (it is also shorter, which is not a bad thing.